### ISSUE TITLE
If a hot publisher's subscriber does not have an error callback, subsequent subscribers also do not get the error propagated to them. 

### ISSUE THREAD

**Issue by arunim29** [reactor/reactor-core#2537]
I have observed that if one of the subscribers of a hot publisher does not have an error callback like doOnError(), all the subsequent subscribers also do not get the errors. Even their doFinally() does not get called. They seem to be in hung state.

I found an issue in RxJs repo which is very similar to this issue.
https://github.com/ReactiveX/rxjs/issues/2145

## Steps to Reproduce

```java
    ConnectableFlux<Integer> connectableFlux = Flux.just(1, 2).doOnNext(i -> {
			if(i==2) {
				throw new RuntimeException();
			}
		})
				.doOnError(ex -> {
					System.out.println("inside doonerror of parent");
				})
				.map(i -> {
					return i;
				})
				.publish();

		// this subscriber does not care about errors
		connectableFlux
		.map(i -> {
			System.out.println("received 1 " + i);
			return i;
		})
		.doFinally(sig -> {
			System.out.println("inside do finally 1 " + sig);
		}).subscribe();
		
		
		//this one does care about errors
		connectableFlux.doOnNext(i -> {
			System.out.println("received 2 " + i);
		})
		.doOnError(ex -> {
			System.out.println("inside do on error 2");
		})
		.doFinally(sig -> {
			System.out.println("inside do finally 2 " + sig);
		}).subscribe();
		
		connectableFlux.connect();
```

## Actual Behavior

The output of above program is -:

```
received 1 1
received 2 1
inside doonerror of parent
inside do finally 1 onError
```

## Expected Behavior

The output should have been -:

```
received 1 1
received 2 1
inside doonerror of parent
inside do finally 1 onError
inside do on error 2
inside do finally 2 onError

```

* Reactor version(s) used: -> 3.3.8.RELEASE

**Comment by ericbottard** [comment]
After some investigation, it appears this has been fixed in 3.4.x, via 7c9bac7953f33aef0301d63dab840fc3c8c87fe6 if I'm not mistaken.
The errors get logged, but the callbacks are all taken care of.

**Comment by simonbasle** [comment]
This is working "as intended" in 3.3.x given the limitation that errors that reach the end of the line where no `onError` handler is provided in the final subscriber will bubble back up the chain as thrown exceptions. For multicast operators that can mean the interruption of processing for the remaining subscribers.

This is something that we did change in 3.4.0, but as that is a breaking change (log the error only instead of bubble it back) we cannot backport it in 3.3.x.

(actually, much of the discussion in the linked RxJs issue apply here too, this is similar to the v5 vs v6 differences in RxJs)

**Comment by arunim29** [comment]
@simonbasle I see that the issue has been fixed in 3.4.x. But in 3.3.x, as this is a publish() operator, shouldn't exceptions coming from the parent or even any of the subscriber be propagated to other subscribers as the other subscribers will anyways stop working bcoz one subscriber(where exeption is thrown) has demand of 0.
